### ISSUE TITLE
llava-intern2-7b模型无法复现官方结果

### ISSUE THREAD

**Issue by PeterMao11** [InternLM/xtuner#346]
模型训练loss:
<img width="1175" alt="image" src="https://github.com/InternLM/xtuner/assets/26111139/5974d260-b6bd-4a6c-82d3-8937f19a99be">

训练脚本：NPROC_PER_NODE=8 xtuner train llava_internlm2_chat_7b_qlora_clip_vit_large_p14_336_lora_e1_gpu8_finetune --deepspeed deepspeed_zero2

模型convert问题：
模型转换OOM：
NPROC_PER_NODE=8 xtuner convert pth_to_hf $FINETUNE_CFG $PTH_PATH $SAVE_PATH
在40G-A100出现OOM情况

NPROC_PER_NODE=1 xtuner convert pth_to_hf $FINETUNE_CFG $PTH_PATH $SAVE_PATH
可以进行模型转换

mmbench-dev评测结果：

┏━━━━━━━━━━┳━━━━━━━━━━━━━━┓
┃ Category ┃ Accuracy (%) ┃
┡━━━━━━━━━━╇━━━━━━━━━━━━━━┩
│ Average  │         68.6 │
├──────────┼──────────────┤
│ AR       │         68.8 │
│ CP       │         83.1 │
│ FP-C     │         57.3 │
│ FP-S     │         69.3 │
│ LR       │         44.9 │
│ RR       │         67.0 │

**Comment by LZHgrla** [comment]
convert 命令无需传递 `NPROC_PER_NODE=xxxx` 参数，直接`xtuner convert` 即可

**Comment by PeterMao11** [comment]
> iteration数为25954有一点奇怪，请问您 config 的 batchsize 是多少？
 iteration以下图为准，batchsize:4 8卡A100
<img width="1165" alt="image" src="https://github.com/InternLM/xtuner/assets/26111139/52ffce5c-fb71-4635-90d1-4444b42e7a18">

**Comment by LZHgrla** [comment]
若要复现结果，建议使用与所提供的config所一致的batchsize，即 16，8卡

**Comment by PeterMao11** [comment]
训练GPU 8卡A100-40G，zero2训练吗

**Comment by PeterMao11** [comment]
> convert 命令无需传递 `NPROC_PER_NODE=xxxx` 参数，直接`xtuner convert` 即可

直接使用xtuner convert，会8卡加载并出现OOM情况
<img width="1126" alt="image" src="https://github.com/InternLM/xtuner/assets/26111139/1532b46c-5828-44ab-8bd1-5cb0124f3fb0">
<img width="1177" alt="image" src="https://github.com/InternLM/xtuner/assets/26111139/ff50b6ee-1087-43e4-b966-6dfbca7ce2ba">

**Comment by LZHgrla** [comment]
@PeterMao11  我们是 8卡 A100-80G，zero2 训练。
如果40GB训练报OOM的话，可以考虑将batchsize调整8，accumulative_counts则调整为2，以确保global batchsize一致

**Comment by LZHgrla** [comment]
> > convert 命令无需传递 `NPROC_PER_NODE=xxxx` 参数，直接`xtuner convert` 即可
> 
> 直接使用xtuner convert，会8卡加载并出现OOM情况 <img alt="image" width="1126" src="https://private-user-images.githubusercontent.com/26111139/298490304-1532b46c-5828-44ab-8bd1-5cb0124f3fb0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5MDcwMjUsIm5iZiI6MTcwNTkwNjcyNSwicGF0aCI6Ii8yNjExMTEzOS8yOTg0OTAzMDQtMTUzMmI0NmMtNTgyOC00NGFiLThiZDEtNWNiMDEyNGYzZmIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIyVDA2NTg0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFiOWY4YjFlOTY2MDAyMzY2Mzc5MjY3M2Q0NWRmOTZjN2Y4NGYwNGZjMDllMmJiOTQ1NmM0ZjQwYmNlYjUxN2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.EGmMZ2vQdfihNY1bdNKNgDXxhsDoim0PEBKnZ5_MLGw"> <img alt="image" width="1177" src="https://private-user-images.githubusercontent.com/26111139/298490478-ff50b6ee-1087-43e4-b966-6dfbca7ce2ba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5MDcwMjUsIm5iZiI6MTcwNTkwNjcyNSwicGF0aCI6Ii8yNjExMTEzOS8yOTg0OTA0NzgtZmY1MGI2ZWUtMTA4Ny00M2U0LWI5NjYtNmRmYmNhN2NlMmJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIyVDA2NTg0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY4NmRjNTBhZmQzYTgxOTljYmYyMjE5YTZhNDA4Mjk1OWQ2Y2UzYzAzYjI2MTEwNzE5NGRjZGIwYzY0ZTdkNTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5HGB-UMs7Y3SDOa889FV2dw_gI_6Af2nEhatLybausY">

convert时，不要使用 NPROC_PER_NODE=xxxx 参数

**Comment by PeterMao11** [comment]
> > > convert 命令无需传递 `NPROC_PER_NODE=xxxx` 参数，直接`xtuner convert` 即可
> > 
> > 
> > 直接使用xtuner convert，会8卡加载并出现OOM情况 <img alt="image" width="1126" src="https://private-user-images.githubusercontent.com/26111139/298490304-1532b46c-5828-44ab-8bd1-5cb0124f3fb0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5MDcwMjUsIm5iZiI6MTcwNTkwNjcyNSwicGF0aCI6Ii8yNjExMTEzOS8yOTg0OTAzMDQtMTUzMmI0NmMtNTgyOC00NGFiLThiZDEtNWNiMDEyNGYzZmIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIyVDA2NTg0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFiOWY4YjFlOTY2MDAyMzY2Mzc5MjY3M2Q0NWRmOTZjN2Y4NGYwNGZjMDllMmJiOTQ1NmM0ZjQwYmNlYjUxN2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.EGmMZ2vQdfihNY1bdNKNgDXxhsDoim0PEBKnZ5_MLGw"> <img alt="image" width="1177" src="https://private-user-images.githubusercontent.com/26111139/298490478-ff50b6ee-1087-43e4-b966-6dfbca7ce2ba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5MDcwMjUsIm5iZiI6MTcwNTkwNjcyNSwicGF0aCI6Ii8yNjExMTEzOS8yOTg0OTA0NzgtZmY1MGI2ZWUtMTA4Ny00M2U0LWI5NjYtNmRmYmNhN2NlMmJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIyVDA2NTg0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY4NmRjNTBhZmQzYTgxOTljYmYyMjE5YTZhNDA4Mjk1OWQ2Y2UzYzAzYjI2MTEwNzE5NGRjZGIwYzY0ZTdkNTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5HGB-UMs7Y3SDOa889FV2dw_gI_6Af2nEhatLybausY">
> 
> convert时，不要使用 NPROC_PER_NODE=xxxx 参数

不使用 NPROC_PER_NODE=1 会出现OOM情况， NPROC_PER_NODE=1才可以正常convert。这里convert是否会影响模型评测结果

**Comment by LZHgrla** [comment]
> > > > convert 命令无需传递 `NPROC_PER_NODE=xxxx` 参数，直接`xtuner convert` 即可
> > > 
> > > 
> > > 直接使用xtuner convert，会8卡加载并出现OOM情况 <img alt="image" width="1126" src="https://private-user-images.githubusercontent.com/26111139/298490304-1532b46c-5828-44ab-8bd1-5cb0124f3fb0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5MDcwMjUsIm5iZiI6MTcwNTkwNjcyNSwicGF0aCI6Ii8yNjExMTEzOS8yOTg0OTAzMDQtMTUzMmI0NmMtNTgyOC00NGFiLThiZDEtNWNiMDEyNGYzZmIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIyVDA2NTg0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFiOWY4YjFlOTY2MDAyMzY2Mzc5MjY3M2Q0NWRmOTZjN2Y4NGYwNGZjMDllMmJiOTQ1NmM0ZjQwYmNlYjUxN2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.EGmMZ2vQdfihNY1bdNKNgDXxhsDoim0PEBKnZ5_MLGw"> <img alt="image" width="1177" src="https://private-user-images.githubusercontent.com/26111139/298490478-ff50b6ee-1087-43e4-b966-6dfbca7ce2ba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5MDcwMjUsIm5iZiI6MTcwNTkwNjcyNSwicGF0aCI6Ii8yNjExMTEzOS8yOTg0OTA0NzgtZmY1MGI2ZWUtMTA4Ny00M2U0LWI5NjYtNmRmYmNhN2NlMmJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIyVDA2NTg0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY4NmRjNTBhZmQzYTgxOTljYmYyMjE5YTZhNDA4Mjk1OWQ2Y2UzYzAzYjI2MTEwNzE5NGRjZGIwYzY0ZTdkNTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5HGB-UMs7Y3SDOa889FV2dw_gI_6Af2nEhatLybausY">
> > 
> > 
> > convert时，不要使用 NPROC_PER_NODE=xxxx 参数
> 
> 不使用 NPROC_PER_NODE=1 会出现OOM情况， NPROC_PER_NODE=1才可以正常convert。这里convert是否会影响模型评测结果

"不使用 NPROC_PER_NODE=1 会出现OOM情况" 这个就很奇怪，你的log看起来是设置了NPROC_PER_NODE=8，所以请查看一下是否export设置了环境变量`NPROC_PER_NODE`

**Comment by davidluciolu** [comment]
您好，请问推理一遍mmbench dev en需要多久？ @PeterMao11

**Comment by StarCycle** [comment]
@davidluciolu 您可以参考#376
